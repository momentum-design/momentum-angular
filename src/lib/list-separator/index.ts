export * from './list-separator.module';
export * from './list-separator.component';
