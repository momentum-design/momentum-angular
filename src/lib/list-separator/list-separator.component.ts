/** @component list-separator */
import { Component, HostBinding, Input } from '@angular/core';

@Component({
  selector: 'cui-list-separator',
  template: `
    <span class="cui-list-separator__container">
      <ng-content *ngIf="!text; else textTemplate"></ng-content>
    </span>

    <ng-template #textTemplate>
      <span
        class="cui-list-separator__text"
        [ngStyle]="{ color: textColor, padding: textPadding }"
        >{{ text }}</span
      >
    </ng-template>
  `,
  host: {
    class: 'cui-list-separator',
  },
})
export class ListSeparatorComponent {
  /** @option Background Color of the ListSeparator | null */
  @Input() backgroundColor: string;
  /** @option Color of the ListSeparator line | null */
  @Input() lineColor: string;
  /** @option Margin of the ListSeparator | null */
  @Input() margin: string;
  /** @option Text of the ListSeparator | null */
  @Input() text: string;
  /** @option TextColor of the ListSeparator | null */
  @Input() textColor: string;
  /** @option Padding around text of the ListSeparator | null */
  @Input() textPadding: string;

  @HostBinding('style.background-color') get _backgroundColor(): string {
    return this.backgroundColor;
  }
  @HostBinding('style.color') get _lineColor(): string {
    return this.lineColor;
  }
  @HostBinding('style.margin') get _margin(): string {
    return this.margin;
  }
}
