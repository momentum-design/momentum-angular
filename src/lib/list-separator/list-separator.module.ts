import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListSeparatorComponent } from './list-separator.component';

@NgModule({
  imports: [CommonModule],
  declarations: [ListSeparatorComponent],
  exports: [ListSeparatorComponent],
})
export class ListSeparatorModule {}
