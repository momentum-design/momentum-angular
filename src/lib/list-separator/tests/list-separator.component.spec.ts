import { Component } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ListSeparatorComponent } from '../list-separator.component';
import { ListSeparatorModule } from '../list-separator.module';

describe('ListSeparatorComponent', () => {
  let testComponent: ListSeparatorComponent;
  let fixture: ComponentFixture<ListSeparatorComponent>;

  @Component({
    selector: 'test-app',
    template: `
      <cui-list-separator class="right">
        <div>Test</div>
      </cui-list-separator>
    `,
  })
  class TestAppComponent { }

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TestAppComponent],
      imports: [ListSeparatorModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ListSeparatorComponent);
    testComponent = fixture.componentInstance;
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(testComponent).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render one ListSeparator', () => {
    fixture.detectChanges();
    expect(fixture.nativeElement.className).toContain('cui-list-separator');
  });

  it('should render text prop', () => {
    testComponent.text = 'Today';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.cui-list-separator__text');
    expect(element).not.toBeNull();
    expect(element.textContent).toEqual('Today');
  });

  it('should change the background color', () => {
    testComponent.backgroundColor = 'transparent';
    fixture.detectChanges();
    expect(fixture.nativeElement.style['background-color']).toEqual('transparent');
  });

  it('should change the line color', () => {
    testComponent.lineColor = 'red';
    fixture.detectChanges();
    expect(fixture.nativeElement.style.color).toEqual('red');
  });

  it('should add padding to the text prop', () => {
    testComponent.text = 'Today';
    testComponent.textPadding = '40px';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.cui-list-separator__text');
    expect(element.style.padding).toEqual('40px');
  });

  it('should change the text color', () => {
    testComponent.text = 'Today';
    testComponent.textColor = 'green';
    fixture.detectChanges();
    const element = fixture.nativeElement.querySelector('.cui-list-separator__text');
    expect(element.style.color).toEqual('green');
  });

  it('should handle class prop', () => {
    const fixtureApp = TestBed.createComponent(TestAppComponent);
    fixtureApp.detectChanges();
    const element = fixtureApp.nativeElement.querySelector('cui-list-separator');
    expect(element.className).toContain('right');
  });

  it('should render children', () => {
    const fixtureApp = TestBed.createComponent(TestAppComponent);
    fixtureApp.detectChanges();
    const element = fixtureApp.nativeElement.querySelector('cui-list-separator');
    expect(element.children.length).toEqual(1);
  });
});
