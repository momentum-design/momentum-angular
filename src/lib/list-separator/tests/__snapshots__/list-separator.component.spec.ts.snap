// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSeparatorComponent should match snapshot 1`] = `
<md-list-separator>
  <span
    class="md-list-separator__container"
  >
    
    
  </span>
</md-list-separator>
`;
